[ADMIN] Adding an MDAnalysis AI tools policy. (#5210)
MDAnalysis/mdanalysis: AI_POLICY.md

Reviewer: Do we want to date or version this policy?

Author: I think that's a great idea. I agree with your point that this should be updated over time.

Reviewer: added as suggestion

> # 🤖 MDAnalysis AI-generated contributions policy 🤖

Reviewer: ```suggestion
# 🤖 MDAnalysis AI-generated contributions policy 🤖

Version 1.0 (2026-02-03)

```

Author: Thanks, I pushed a variant of this as a bolded header isntead.

This document defines MDAnalysis' policy regarding AI-generated content.

> 

Reviewer: ```suggestion
All content under the MDAnalysis GitHub organization (https://github.com/MDAnalysis/) is
subject to this policy. 

```

Author: Thanks, I've added a variant of this.

> For the purposes of this document, _content_ is defined as any contribution to MDAnalysis, both code and non-code, including but not limited to:

Reviewer: We should make clear what "any contribution to MDAnalysis" means. I assume we mean "any code in the github.com/MDAnalysis organization".

Reviewer: see suggestion above

```reST
All content under the MDAnalysis GitHub organization (https://github.com/MDAnalysis/) is
subject to this policy. 
```

- Code submitted for inclusion in MDAnalysis
- Documentation of any kind
- Communication content (e.g. issues, discussions, pull request descriptions, and reviews)

> 

Reviewer: I'd add something like the following as a preamble:

```suggestion

We recognize that the use of AI in open source software raises many difficult questions and acknowledge that views, approaches, and technical capabilities are rapidly changing. 
We invite comments and further discussion (via our [discussion channels](https://www.mdanalysis.org/pages/community/)) and expect to periodically review this policy.

```

Reviewer 2: I had the same thought yesterday. We should make clear that the policy will evolve over time, as we learn and the landscape changes.

Reviewer: I'd like my above suggestion (or something similar) included in v1.0.

## Policy overview

MDAnalysis does not accept any substantial uses of AI-generated content in contributions.

> AI tools may be used in limited cases, for example when it is used for small amounts of AI-assisted code (e.g. autocompletion using copillot, cursor, etc...)

Reviewer 2: ```suggestion
AI tools may be used in limited cases, for example when it is used for small amounts of AI-assisted code (e.g. autocompletion using copilot, cursor, etc...)
```

Author: Looks like my "LLM assist" in google docs didn't do a very good job :P

or for language translation purposes. **Such uses must always be declared.**

## Policy details

### 0. AI usage must always be declared

All contributions must be fully transparent about any AI usage. Failure to disclose AI use (e.g. in pull request description details)
may be queried by other developers and may be considered non-compliance with this policy.

If we suspect that AI tools were used as part of a contribution, MDAnalysis developers may ask you for clarification.
Please do not take offense if this happens to you, distinguishing between AI-generated and human-authored content is increasingly difficult!

### 1. Fully generated code is not allowed

> Fully AI-generated contributions, such as content generated by prompting a large language model (e.g. claude code or chatgpt) are not permitted in MDAnalysis.

Reviewer 2: I'd mention agents explicitly here as well. Additionally, it might be useful to add a short line on why for now they are not allowed to(i.e. concerns about licensing etc.).

Author: I'm afraid I'm rather uneducated when it comes to agents etc... do you have some suggested wording for this?

Reviewer: ```suggestion
Fully AI-generated contributions, such as content generated by prompting a large language model (e.g. claude code or chatgpt) or produced by any agentic AI system are not permitted in MDAnalysis.
```

Some limited exceptions are described in sections #2 and #3 for integrated development environments (IDEs) and language translation.

> ### 2. Limited AI assitance exemption: IDEs and autocompletion

Reviewer: ```suggestion
### 2. Limited AI assistance exemption: IDEs and autocompletion
```

> AI tools are now so deeply integrated into many modern developpment environments, and it may be impractical to avoid them entirely.

Reviewer 2: ```suggestion
AI tools are now so deeply integrated into many modern development environments, and it may be impractical to avoid them entirely.
```

We recognise that this is particularly true for modern integrated development environments (IDEs), which may use AI to power features like autocompletion.

In these limited cases of code generation, typically ranging from a single variable to a couple of lines of code,

> AI assitance is deemed acceptable. However, if code generate exceeds minimal, sporadic amounts (e.g. repeated or large multi-line blocks),

Reviewer 3: generate -> generation?

it would be considered fully AI-generated and, as defined in section #1, is not acceptable.

> As per section #0, where possible please state that you are using AI assistance via an IDE.

Reviewer: Remove "where possible" as this puts the burden on us. Instead, it should be a contributors responsibility to know when they are using AI features.
```suggestion
As per section #0, please state when you are using AI assistance via an IDE.
```

Author: With IDEs, my main worry is that it's just easy to forget - like I'm a vim guy, but sometimes I use vscode and then the auto complete happens but I forget about it.

Reviewer: It's still on the user to know. I don't think the policy is the place to say "we know, you're human, so we are putting qualifiers on our requirements". We should say what we want to happen.

A checklist will make this easier.

Author: I think this may be something to discuss on Friday. For now, I have pushed a different variant of that text which hopefully may offer a middle ground.

> ### 3. Limited AI assitance exemption: language translation

Reviewer: ```suggestion
### 3. Limited AI assistance exemption: language translation
```

MDAnalysis aims to foster an inclusive environment for all contributors and users. We recognise that language barrier can limit participation
in open-source projects, particularly for non-native English speakers.

> Whilst we you to communicate to us in your own words (and languages if needed), we acknowledge that AI-based translation tools may sometimes

Reviewer 2: ```suggestion
Whilst we wish you to communicate to us in your own words (and languages if needed), we acknowledge that AI-based translation tools may sometimes
```
?

> Whilst we wish you to communicate to us in your own words (and languages if needed), we acknowledge that AI-based translation tools may sometimes

Reviewer: English is not my first language so take my following suggestion just as a suggestion:
```suggestion
Whilst we prefer you communicate with us in your own words (and languages if needed), we acknowledge that AI-based translation tools may sometimes
```

be necessary for communicating in issues / discussions. If such tools are used, we ask you to disclose their use and purpose.

### 4. Contributors are responsible authors and must be aware of, and declare, any derivate work

Contributors are responsible for the content and provenance of every aspect of their contribution, including any design decisions.
Contributors must be aware of whether their work derives from external sources / codebases and must declare any derivative work.

### 5. Human reviewers are required

All code merged into MDAnalysis must be reviewed by a human reviewer. Instructions / suggestions from human reviewers always take precedence over those of non-human reviewers.

### 6. Consequences for failure to comply to this policy

> Any failures to comply with this policy should be reported to the MDAnalysis core developers.

Reviewer: Do we need to link to contact information?

Author: Yeah the whole thing needs links, and then itself being linked in other locations. I'm hoping that to be a task once we've agreed on the text.

Reviewer: ```suggestion
Any failures to comply with this policy should be reported to the MDAnalysis core developers through
our our [standard communications channels](https://www.mdanalysis.org/pages/community/#ask-questions--get-help) or email to mdanalysis@numfocus.org .
```

Reviewer: Or should we use licensing@mdanalysis.org ?

Author: the main inbox is fine imho

The core developers may enforce this policy through a range of actions, which may include restricting or banning individuals from future contributions.

> There is a zero tolerance policy for intentially failing to disclose or misrepresenting the extent of AI usage.

Reviewer 3: intentially -> intentionally?

# 🤖 MDAnalysis AI-generated contributions policy 🤖

### Version: 1.0 (2026-02-03)

This document defines the [MDAnalysis organization's](https://github.com/MDAnalysis/) policy regarding AI-generated content. This policy applies to all aspects of the MDAnalysis project, including all the GitHub repositories under the organization.

For the purposes of this document, _content_ is defined as any contribution to the MDAnalysis organization, both code and non-code, including but not limited to:

> - Code submitted for inclusion in MDAnalysis

Reviewer 2: ```suggestion
- Code submitted for inclusion in MDAnalysis, or other organization-owned repositories
```

it would be considered fully AI-generated and, as defined in section #1, is not acceptable.

As per section #0, it is your responsibility to report the extent of AI assitance you used.

### 3. Limited AI assistance exemption: language translation

MDAnalysis aims to foster an inclusive environment for all contributors and users. We recognise that language barrier can limit participation
in open-source projects, particularly for non-native English speakers.

Whilst we prefer you communicate with us in your own words (and languages if needed), we acknowledge that AI-based translation tools may sometimes
be necessary for communicating in issues / discussions. If such tools are used, we ask you to disclose their use and purpose.

### 4. Contributors are responsible authors and must be aware of, and declare, any derivate work

Contributors are responsible for the content and provenance of every aspect of their contribution, including any design decisions.
Contributors must be aware of whether their work derives from external sources / codebases and must declare any derivative work.

### 5. Human reviewers are required

> All code merged into MDAnalysis repositories must be reviewed by a human reviewer. Instructions / suggestions from human reviewers always take precedence over those of non-human reviewers.

Reviewer 2: Should we suggest/ask of not adding Copilot as a reviewer, or that is something we would be OK having?

Author: I don't have a strong view on this, personally folks can do what they want, I just don't want to be second guessed by AI.

### 6. Consequences for failure to comply to this policy

Any failures to comply with this policy should be reported to the MDAnalysis core developers through

> our our [standard communications channels](https://www.mdanalysis.org/pages/community/#ask-questions--get-help) or email to mdanalysis@numfocus.org .

Reviewer 2: ```suggestion
our [standard communications channels](https://www.mdanalysis.org/pages/community/#ask-questions--get-help) or email to mdanalysis@numfocus.org .
```
